### PR TITLE
python_requirements_info - improvements

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Returns python version broken down into its components, and some minor refactoring.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/modules/system/python_requirements_info.py

### CODE REVIEW

File: plugins/modules/system/python_requirements_info.py [hunk 1]
```diff
@@ -106,11 +115,28 @@
...
+        releaselevel=sys.version_info.releaselevel,
+        serial=sys.version_info.serial,
+    )
+
 
 def main():
     module = AnsibleModule(
         argument_spec=dict(
-            dependencies=dict(type='list', elements='str')
+            dependencies=dict(type='list', elements='str', default=[])
```
Comment (felixfontein) [inline plugins/modules/system/python_requirements_info.py:139]: I'm somewhat surprised that the sanity tests aren't screaming. The documentation does not say there is a default.

Comment (russoz) [reply plugins/modules/system/python_requirements_info.py:139]: I wonder if the sanity tests check are checking this in a boolean-ish way. Fixing docs now.

File: plugins/modules/system/python_requirements_info.py [hunk 2]
```diff
@@ -48,14 +48,23 @@
...
+python_version_info:
+  description: breakdown version of python
+  returned: always
+  type: dict
+  sample:
+    major: 3
+    minor: 8
+    micro: 10
+    releaselevel: final
+    serial: 0
```
[the remark this answers is not in the document]
Comment (russoz) [reply plugins/modules/system/python_requirements_info.py]: fixed

File: plugins/modules/system/python_requirements_info.py (hunk 1 above)
Comment (russoz) [reply plugins/modules/system/python_requirements_info.py:152]: fixed

**Review by felixfontein — APPROVED** [review]

**Pull request merged**